### PR TITLE
Check that the script is not running in a Cygwin shell -- the symboli…

### PR DESCRIPTION
**Pull request by nmatthews-witekio**
…c linking doesn't work.

Check if wget or curl are present and modify FETCH command accordingly. Note: Windows git bash shell has curl but not wget.

### CODE REVIEW

File: scripts/setup-project.sh [hunk 1]
```diff
@@ -2,6 +2,27 @@
 
 set -e
 
+if [ $(uname -o | grep "Cygwin") ]; then
+  echo "Warning: this script does not support Cygwin. Please use WSL or 'git bash'."
```
Comment (rbateman-witekio) [inline scripts/setup-project.sh]: Is it clear why the symlinks don't work under cygwin? It could be that we just don't check for it in make_sym_link

Comment (nmatthews-witekio) [reply scripts/setup-project.sh]: e2studio doesn't seem to deal with Cygwin soft links well.
Cygwin doesn't have mklink (at least in a default setup).

I just thought it was safer to give a warning and stop, than pretend that everything was ok.

Comment (rbateman-witekio) [reply scripts/setup-project.sh]: For WSL and Msys2 we actually invoke the Windows shell to create the link. Can cygwin do the same? Shouldn't be too much work to add a check for cygwin in make_sym_link

**Review by nik-markovic — COMMENTED** [review]
What do you think about just switching to curl instead of wget, to simplify the whole thing? Is curl avaialble in msys2 or git bash by default? Where is wget not avaialble by default?

What we have currently is fine, but just laying it out.

**Comment by nmatthews-witekio** [comment]
Well, I've found one system that only has curl -- I guess there might be one that only has wget? So, I'd just leave both in at present.

**Comment by nik-markovic** [comment]
> 

I'm thinking just use curl to simplify it... I tested with msys2 and git bash  they both **have** curl built-in.

I'm approving the intended change, but up to you if you feel we can simplify.

Edit: Sorry accidentally closed the PR. I reopened it.

File: scripts/setup-project.sh [hunk 2]
```diff
@@ -2,6 +2,22 @@
...
 set -e
 
+#
+# Select download mechanism given abilities of shells, etc.
+#
+which wget >& /dev/null
+if [ $? == 0 ]; then
+  FETCH="wget -q -O"
+else
+  which wget >& /dev/null
```
Comment (nik-markovic) [inline scripts/setup-project.sh]: Looks like a copy-paste typo. You wanted to check **which curl** probably at line 12?

**Review by nik-markovic — CHANGES_REQUESTED** [review]

**Review by nik-markovic — APPROVED** [review]

**Pull request merged**